sdks/python/apache_beam/transforms/enrichment.py

```diff
@@ -15,47 +15,48 @@
 # limitations under the License.
 #
 
+from typing import Any
 from typing import Callable
-from typing import Generic
+from typing import Dict
 from typing import Optional
+from typing import Tuple
 from typing import TypeVar
 
 import apache_beam as beam
 from apache_beam.io.requestresponse import DEFAULT_TIMEOUT_SECS
-from apache_beam.io.requestresponse import CacheReader
-from apache_beam.io.requestresponse import CacheWriter
 from apache_beam.io.requestresponse import Caller
-from apache_beam.io.requestresponse import PreCallThrottler
-from apache_beam.io.requestresponse import Repeater
 from apache_beam.io.requestresponse import RequestResponseIO
-from apache_beam.io.requestresponse import ShouldBackOff
 
 __all__ = [
     "EnrichmentSourceHandler",
     "Enrichment",
+    "cross_join",
 ]
 
 InputT = TypeVar('InputT')
 OutputT = TypeVar('OutputT')
 
+JoinFn = Callable[[Tuple[Dict[str, Any], Dict[str, Any]]], beam.Row]
```

Comment:
Nit: Since we're exposing this to users, it's probably better to make the JoinFn of the form:

`def join_fn(left: Dict[str, Any], right: Dict[str, Any]) -> beam.Row:`

That's easier to understand and doesn't force every user to repeat the `left, right = element` line. This also then requires flipping the map below to:

`return fetched_data | beam.Map(lambda x: self._join_fn(x[0], x[1]))`

Reply:
I agree, thanks!